agent/api/container/container.go

```diff
@@ -1564,16 +1564,9 @@ func (c *Container) GetImageName() string {
 
 // Checks if the container has a resolved image manifest digest.
 // Always returns false for internal containers as those are out-of-scope of digest resolution.
+// Always returns false when container's image reference contains digest as no digest resolution is needed in that case.
 func (c *Container) DigestResolved() bool {
-	return !c.IsInternal() && c.GetImageDigest() != ""
-}
-
-// Checks if the container's image requires manifest digest resolution.
-// Manifest digest resolution is required if the container's image reference does not
-// have a digest.
-// Always returns false for internal containers as those are out-of-scope of digest resolution.
-func (c *Container) DigestResolutionRequired() bool {
-	return !c.IsInternal() && referenceutil.GetDigestFromImageRef(c.Image) == ""
+	return !c.IsInternal() && c.GetImageDigest() != "" && !referenceutil.DigestExists(c.Image)
```

Comment:
`!referenceutil.DigestExists(c.Image)` When do we hit this case, and why do we need the `not` in front of it?

Reply:
for example, if a task run with image`ubuntu@sha256:ed6d2c43c8fbcd3eaa44c9dab6d94cb346234476230dc1681227aa72d07181ee` then `referenceutil.DigestExists(c.Image)` is true since image reference already contains digest,  so no digest resolution is needed in this case,  and we don't need to make an additional STSC call to report digest to backend, so we add `not` in front of it to prevent this container from marking as digest resolved.  

if the task has at least one container with `DigestResolved` then the state change will be sent to the ECS backend (additional STSC call)